custom-checks/pmd/src/main/java/org/openhab/tools/analysis/pmd/UseSLF4JLoggerRule.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.tools.analysis.pmd;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import net.sourceforge.pmd.lang.ast.Node;
+import net.sourceforge.pmd.lang.java.ast.ASTClassOrInterfaceType;
+import net.sourceforge.pmd.lang.java.ast.ASTImportDeclaration;
+import net.sourceforge.pmd.lang.java.ast.ASTReferenceType;
+import net.sourceforge.pmd.lang.java.ast.ASTType;
+import net.sourceforge.pmd.lang.java.ast.ASTVariableDeclarator;
+import net.sourceforge.pmd.lang.java.rule.AbstractJavaRule;
+
+/**
+ * Checks if a logger other than the one provided by slf4j is used
+ *
+ * @author Lyubomir Papazov - Initial contribution
+ *
+ */
+public class UseSLF4JLoggerRule extends AbstractJavaRule {
+
+
+    private static final String LOGGER_LITERAL = "Logger";
+
+    private Set<String> forbiddenLoggers = new HashSet<String>();
+    private boolean isSlf4jPackageImported;
+
+    public UseSLF4JLoggerRule() {
+        forbiddenLoggers.add("org.apache.log4j.Logger");
+        forbiddenLoggers.add("java.util.logging.Logger");
+        forbiddenLoggers.add("ch.qos.logback.classic.Logger");
+        forbiddenLoggers.add("org.apache.commons.logging.Log");
+    }
+
+    @Override
+    public Object visit(ASTImportDeclaration node, Object data) {
+        String fullImportName = node.getImportedName();
+        if (forbiddenLoggers.contains(fullImportName)) {
+            addViolation(data, node);
+        } else if (fullImportName.equals("org.slf4j.Logger") || (fullImportName.equals("org.slf4j") && node.isImportOnDemand())) {
+            isSlf4jPackageImported = true;
+        }
+        return super.visit(node, data);
+    }
+
+    @Override
+    public Object visit(ASTVariableDeclarator node, Object data) {
+        ASTType typeNode = node.jjtGetParent().getFirstChildOfType(ASTType.class);
+        Node reftypeNode = typeNode.jjtGetChild(0);
+        if (reftypeNode instanceof ASTReferenceType) {
+            ASTClassOrInterfaceType classOrIntType = reftypeNode.getFirstChildOfType(ASTClassOrInterfaceType.class);
```

Comment:
Why not `classOrInterfaceType` you had me confused for 5 minutes what an `int` would be doing here...